packages/@aws-cdk/aws-apigateway/lib/usage-plan.ts

```diff
@@ -0,0 +1,241 @@
+import { IRestApi } from './restapi';
+import { Stage } from './stage';
+import { validateInteger } from './util';
+
+/**
+ * Container for defining throttling parameters to API stages or methods.
+ * @link https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-request-throttling.html
+ */
+export interface ThrottleSettings {
+  /**
+   * The API request steady-state rate limit (average requests per second over an extended period of time)
+   *
+   * Type: Integer
+   */
+  readonly rateLimit?: number;
+
+  /**
+   * The maximum API request rate limit over a time ranging from one to a few seconds.
+   *
+   * Type: Integer
+   */
+  readonly burstLimit?: number;
+}
+
+/**
+ * Time period for which quota settings apply.
+ */
+export enum Period {
+  Day = 'DAY',
+  Week = 'WEEK',
+  Month = 'MONTH'
+}
+
+/**
+ * Specifies the maximum number of requests that clients can make to API Gateway APIs.
+ */
+export interface QuotaSettings {
+  /**
+   * The maximum number of requests that users can make within the specified time period.
+   *
+   * Type: Integer
+   */
+  readonly limit?: number;
+
+  /**
+   * For the initial time period, the number of requests to subtract from the specified limit.
+   *
+   * Type: Integer
+   */
+  readonly offset?: number;
+
+  /**
+   * The time period for which the maximum limit of requests applies.
+   */
+  readonly period?: Period;
+}
+
+/**
+ * Represents per-method throttling for a resource.
+ */
+export interface ThrottlingPerMethod {
+  /**
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly method: Method,
+  readonly throttle: ThrottleSettings
+}
+
+/**
+ * Type of Usage Plan Key. Currently the only supported type is 'API_KEY'
+ */
+export enum UsagePlanKeyType {
+  ApiKey = 'API_KEY'
+}
+
+/**
+ * Represents the API stages that a usage plan applies to.
+ */
+export interface UsagePlanPerApiStage {
+
+  readonly api?: IRestApi,
+
+  /**
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly stage?: Stage,
+  readonly throttle?: ThrottlingPerMethod[]
+}
+
+export interface UsagePlanProps {
+  /**
+   * API Stages to be associated which the usage plan.
+   */
+  readonly apiStages?: UsagePlanPerApiStage[],
+
+  /**
+   * Represents usage plan purpose.
+   */
+  readonly description?: string,
+
+  /**
+   * Number of requests clients can make in a given time period.
+   */
+  readonly quota?: QuotaSettings
+
+  /**
+   * Overall throttle settings for the API.
+   */
+  readonly throttle?: ThrottleSettings,
+
+  /**
+   * Name for this usage plan.
+   */
+  readonly name?: string
+
+  /**
+   * ApiKey to be associated with the usage plan.
+   */
+  readonly apiKey?: IApiKey
+}
+
+export class UsagePlan extends Resource {
+  /**
+   * @attribute
+   */
+  public readonly usagePlanId: string;
+
+  constructor(scope: Construct, id: string, props: UsagePlanProps = { }) {
+    super(scope, id);
+    let resource: CfnUsagePlan;
+
+    if (props !== undefined) {
```

Comment:
Doesn't look like `props` can ever be undefined, so this `if` seems unnecessary.